### DISCUSSION TITLE
Speaker Consistency Loss (SCL) in XTTS

### DISCUSSION THREAD

**Discussion by yonas-g** [idiap/coqui-ai-TTS#487] [Q&A] [answered]
Hi everyone,

I was trying to see if SCL is implemented in the code for finetuning but I couldn't find any loss related to it in the code. The current training code only considers `Cross Entropy` for the text and mel in `layers/xtts/gpt.py` between lines [`469:475`](https://github.com/idiap/coqui-ai-TTS/blob/982c566429c70afa3970d14fee185a8abe0b5823/TTS/tts/layers/xtts/gpt.py#L469). I was wondering whether it has been implemented in the code since I couldn't find any reference.

The XTTS paper mentions
> Our proposed decoder is also conditioned with speaker embedding from the H/ASP model [heo2020clova](https://arxiv.org/html/2406.04904v1#bib.bib24). The speaker embedding was added in each upsampling layer via linear projection. Inspired by [yourtts](https://arxiv.org/html/2406.04904v1#bib.bib4), to improve the speaker similarity, we also added the Speaker Consistency Loss (SCL).

While I can confirm that the speaker embedding is added in the HIFIGAN decoder. There is no SCL loss in the code.

**Comment by eginhard** [comment, marked as answer]
The repository only has the inference code for the HiFi-GAN decoder. The training code that would contain this loss term was never released.

**Reply by yonas-g** [reply]
True, there is no training code for the HiFi-GAN decoder in the code. In addition, the XTTS training code in the repo also misses additional auxiliary loss (SCL) which I believe was used in the original XTTS training as mentioned in the paper.

Interestingly, a model trained without it causes the model not to adhere to the reference audio provided during if you use very large and diverse dataset to finetune. It could also be the fact that the dataset I am using is OOD for the speaker encoder model.
